app/uk/gov/hmrc/individualincomedesstub/repository/SelfEmploymentRepository.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualincomedesstub.repository
+
+import javax.inject.{Inject, Singleton}
+
+import play.api.libs.json.Json
+import reactivemongo.api.indexes.Index
+import reactivemongo.api.indexes.IndexType.Ascending
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json._
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.individualincomedesstub.domain._
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class SelfEmploymentRepository @Inject()(mongoConnectionProvider: MongoConnectionProvider)
+  extends ReactiveRepository[SelfEmployment, BSONObjectID]("selfEmployment", mongoConnectionProvider.mongoDatabase, JsonFormatters.selfEmploymentFormat) {
+
+  override lazy val indexes = Seq(
+    Index(key = Seq(("nino", Ascending)), name = Some("ninoIndex"), unique = false, background = true)
+  )
+
+  def create(nino: Nino, taxYear: TaxYear, request: CreateSelfEmploymentRequest) = {
```

Comment:
I think better to pass in SelfEmployment in method here and convert CreateSelfEmploymentRequest to SelfEmployment in the service